File: DesignPatternsDotNet/MementoPattern/CareTaker.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections;

namespace MementoPattern
{
    class CareTaker
    {
        private ArrayList mementos = new ArrayList();
        public void ADD(Memento aMemento)
        {
            mementos.Add(aMemento);
            Console.WriteLine("One Item added, so no item count : " +mementos.Count);
        }
        public Memento GetMemento(int index)
        {
            return (Memento)mementos[index];
        }
    }
}

File: DesignPatternsJava/src/main/java/com/memento/CareTaker.java
package com.memento;

import java.util.ArrayList;

public class CareTaker {
	ArrayList<Memento> mementoes = new ArrayList<Memento>();
	public void add(Memento aMemento){
		mementoes.add(aMemento);
		System.out.println("One Item added, so no item count : "+mementoes.size());
	}
	public Memento getMemento(int index) {
		return mementoes.get(index);		
	}
}
